### ISSUE TITLE
I couldn't get this to actually work! :(

### ISSUE THREAD

**Issue by yaronyg** [DrJukka/BTConApp#2]
I am using a Samsung S5 running Android 4.3 and a Nexus 7 running 4.4.4. The apps start just fine, gives the 'hi there' and show finding peers.

The Samsung log is [here](https://gist.github.com/yaronyg/5cb77af9aa43d03523da) and the Nexus 7 log is available [here](https://gist.github.com/yaronyg/9a67b4a4718257aa5b37).

The only thing that stood out to me is:

```
W/BluetoothAdapter﹕ getBluetoothService() called with no BluetoothManagerCallback
```

Help!

**Comment by DrJukka** [comment]
Will need to investigate. The Samsung appears to get new events for the Bluetooth mode all the time, which makes it re-start the services (since it though that BT was just turned  back on)

The Nexus, simply never gets any devices discovered, thus its timing out the Peer discovery, and re-starting searching the devices.

**Comment by yaronyg** [comment]
Since I only have two Android devices it makes sense that the Nexus isn't finding anything since the Samsung is having a bad day. What can we do about the Samsung?

**Comment by DrJukka** [comment]
I was doing a bit of updates to the code today. Noticed that my Samsung did not generate events for BT On/off, but now it handles the situation loads better (i.e. no crashing). Also got both my Nexus 4 devices into a state where they try finding peers, and are not finding any, as well as they were invisible to others while being confuced.

Swithing Wlan off, and back on fixed the issue. Generally, re-booting device (if reset is out of option) should normally fix any issue

**Comment by yaronyg** [comment]
Yes but shipping a library that tells the user 'hey, when you have a problem, reboot'. Isn't going to win us friends. Why did things fail in the first place?

**Comment by DrJukka** [comment]
Af far as I can see, the API sucks sometimes. I'll be publishing some data on issues soon. Basically there is repeatable issue, in which the device stops finding services, and that have happened with all devices so far. The peers being lost, so far only seen with Nexus 4. Will see if there is better ways, but from code perspective, its as simply as me calling to find peers, and never getting any event for finding one. Next event I get is discovery stopping, and the logical thing there to do, is to restart the peer discovery. 

Though, gottta say, need to investigate and see if there is anything else we could do.
